openedx/core/djangoapps/profile_images/views.py

```diff
@@ -0,0 +1,96 @@
+"""
+"""
+
+from contextlib import closing
+
+from rest_framework import permissions, status
+from rest_framework.authentication import OAuth2Authentication, SessionAuthentication
+from rest_framework.parsers import MultiPartParser, FormParser
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from ..user_api.accounts.api import set_has_profile_image, get_profile_image_names
+
+from .images import validate_uploaded_image, generate_profile_images, remove_profile_images, ImageFileRejected
+
+
+class ProfileImageUploadView(APIView):
+    """
+    Provides a POST endpoint to generate new profile image files for a given
+    user, using an uploaded source image.
+    """
+
+    parser_classes = (MultiPartParser, FormParser,)
+
+    authentication_classes = (OAuth2Authentication, SessionAuthentication)
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def post(self, request, username):
+        """
+        HTTP POST handler.
+        """
+
+        # validate request:
+        # ensure authenticated user is either same as username, or is staff.
+        if request.user.username != username and not request.user.is_staff:
```

Comment:
Works for me.  There is one thing that feels a little weird which is that the staff user will get 404s - it seems like their elevated permission should at least entitle them to a 403!  Do you think it would be worth patching that permission to give 403s in that case?